### PR TITLE
[v6.0] Accumulate ErgoTree deserialization cost

### PR DESCRIPTION
**Pull request by SethDusek**
For #846 

I've accumulated the deserialization cost into the total cost and updated all the relevant test cases with costs for all ErgoTree versions when V3 is activated.

### CODE REVIEW

**Review by kushti — APPROVED** [review]

**Comment by SethDusek** [comment]
Please re-run CI, I believe this CI failure was due to an issue in the underlying branch and not this PR itself. Github doesn't seem to automatically re-run CI for PRs when the base branch changes

**Comment by aslesarenko** [comment]
> Github doesn't seem to automatically re-run CI for PRs when the base branch changes

Yes, because changing the base branch doesn't change the commit which is tested. 
Also, "re-run" command in somehow not available in the logs.

**Comment by kushti** [comment]
Indeed, ErgoTreeSerializerSpecification is failing, which is unlikely touched by this PR and coming from upstream

**Comment by kushti** [comment]
@aslesarenko please review

File: sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala [hunk 1]
```diff
@@ -266,44 +266,45 @@ class ScriptVersionSwitchSpecification extends SigmaDslTesting {
 
   /** Rule#| BlockVer | Block Type| Script Version | Release | Validation Action
     * -----|----------|-----------|----------------|---------|--------
-    * 19   | 4        | candidate | Script v3      | v5.0    | skip-accept (rely on majority)
-    * 20   | 4        | mined     | Script v3      | v5.0    | skip-accept (rely on majority)
+    * 19   | 4        | candidate | Script v3      | v6.0    | R6.0-JIT-verify
+    * 20   | 4        | mined     | Script v3      | v6.0    | R6.0-JIT-verify
```
Comment (aslesarenko) [inline sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala:270]: this doesn't look right. why Validation Action changed?
The whole set of rules is designed for two versions V and V+1 and describes switch from current version to the next version.
The semantics of each rule should remain the same, but versions should move from v to v+1.

File: sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala [hunk 2]
```diff
@@ -266,44 +266,45 @@ class ScriptVersionSwitchSpecification extends SigmaDslTesting {
...
     */
   property("Rules 19,20 | Block v4 | candidate or mined block | Script v3") {
     forEachActivatedScriptVersion(activatedVers = Array[Byte](3)) // version for Block v4
     {
-      forEachErgoTreeVersion(ergoTreeVers = Array[Byte](3, 4)) { // scripts >= v3
+      forEachErgoTreeVersion(ergoTreeVers = Array[Byte](3)) { // scripts >= v3
```
Comment (aslesarenko) [inline sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala:275]: same is here, versions 3,4 in v5.x means versions higher that MaxSupportedVersion, so should be 4,5

File: sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala [hunk 3]
```diff
@@ -266,44 +266,45 @@ class ScriptVersionSwitchSpecification extends SigmaDslTesting {
...
 
-        // prover is rejecting, because such context parameters doesn't make sense
-        assertExceptionThrown(
-          testProve(ergoTree, activatedScriptVersion = activatedVersionInTests),
-          exceptionLike[InterpreterException](s"Both ErgoTree version ${ergoTree.version} and activated version $activatedVersionInTests is greater than MaxSupportedScriptVersion $MaxSupportedScriptVersion")
-        )
+        // both prove and verify are accepting with full evaluation
+        val pr = testProve(ergoTree, activatedScriptVersion = activatedVersionInTests)
+        pr.proof shouldBe Array.emptyByteArray
+        pr.cost shouldBe 24L
```
Comment (aslesarenko) [inline sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala:282]: again, only versions should change in the rules, but not the rules themselves.

File: sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala [hunk 4]
```diff
@@ -266,44 +266,45 @@ class ScriptVersionSwitchSpecification extends SigmaDslTesting {
...
-    * 24   | 4        | mined     | Script v2      | v5.0    | R5.0-JIT-verify
+    * 21   | 4        | candidate | Script v0/v1   | v5.0    | R6.0-JIT-verify
+    * 22   | 4        | candidate | Script v2      | v5.0    | R6.0-JIT-verify
+    * 23   | 4        | candidate | Script v3      | v5.0    | R6.0-JIT-verify
+    * 24   | 4        | mined     | Script v0/v1   | v5.0    | R6.0-JIT-verify
+    * 25   | 4        | mined     | Script v2      | v5.0    | R6.0-JIT-verify
+    * 26   | 4        | mined     | Script v3      | v5.0    | R6.0-JIT-verify
     */
-  property("Rules 21,22,23,24 | Block v4 | candidate or mined block | Script v0/v1/v2") {
+  property("Rules 21,22,23,24,25,26 | Block v4 | candidate or mined block | Script v0/v1/v2/v3") {
```
Comment (aslesarenko) [inline sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala:301]: new rules are not necessary, instead the changes should be something like this
```
    * 21   | 4        | candidate | Script v0/v1/v2   | v6.0    | R6.0-JIT-verify
    * 22   | 4        | candidate | Script v3      | v6.0    | R6.0-JIT-verify
    * 23   | 4        | mined     | Script v0/v1/v2   | v6.0    | R6.0-JIT-verify
    * 24   | 4        | mined     | Script v3      | v6.0    | R6.0-JIT-verify
```

File: sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala [hunk 5]
```diff
@@ -266,44 +266,45 @@ class ScriptVersionSwitchSpecification extends SigmaDslTesting {
...
     // this test verifies the normal validation action R5.0-JIT-verify of v5.x releases
-    // when Block v4 already activated, but the script is v0, v1 or v2.
+    // when Block v4 already activated, and the script is v0, v1, v2, or v3.
 
     forEachActivatedScriptVersion(Array[Byte](3)) // version for Block v4
     {
-      forEachErgoTreeVersion(Array[Byte](0, 1, 2)) { // tree versions supported by v5.x
+      forEachErgoTreeVersion(Array[Byte](0, 1, 2, 3)) { // tree versions supported by v6.x
```
Comment (aslesarenko) [inline sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala:307]: this test is for the case when Block version is above currently supported version.
I.e. for when  v6.0 release is current version, then the next block version is 5. (i.e. for v7.0)

**Review by aslesarenko — CHANGES_REQUESTED** [review]

File: sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala (hunk 1 above)
Comment (SethDusek) [reply sc/shared/src/test/scala/sigmastate/ScriptVersionSwitchSpecification.scala]: Changed to Block V5 with Script V4 activated

**Pull request merged**